VisualPinball.Unity/VisualPinball.Unity.Editor/DragPointsEditor/DragPointsEditor.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+using VisualPinball.Unity.VPT;
+using VisualPinball.Unity.Extensions;
+using VisualPinball.Engine.Math;
+
+
+namespace VisualPinball.Unity.Editor.DragPoints
+{
+	public class DragPointsEditor
+	{
+		List<DragPointData> controlPoints = new List<DragPointData>();
+
+        List<Vector3> pathPoints = new List<Vector3>();
+
+		public void OnSceneGUI(Object target)
+		{
+			IEditableItemBehavior editable = target as IEditableItemBehavior;
+			IDragPointsEditable dpeditable = target as IDragPointsEditable;
+			Behaviour bh = target as Behaviour;
+
+			if (bh == null || dpeditable == null)
+				return;
+
+			controlPoints.Clear();
+
+			Vector3 offset = dpeditable.GetEditableOffset();
+			Matrix4x4 lwMat = bh.transform.localToWorldMatrix;
+			Matrix4x4 wlMat = bh.transform.worldToLocalMatrix;
+
+			foreach (var dpoint in dpeditable.GetDragPoints())
+			{
+				Vector3 dpos = dpoint.Vertex.ToUnityVector3();
+				dpos.z += dpoint.CalcHeight;
+				dpos += offset;
+				dpos = lwMat.MultiplyPoint(dpos);
+				EditorGUI.BeginChangeCheck();
+				dpos = Handles.PositionHandle(dpos, Quaternion.identity);
+				DragPointData transformedData = new DragPointData(dpoint);
+				transformedData.Vertex = dpos.ToVertex3D();
+				controlPoints.Add(transformedData);
+				if (EditorGUI.EndChangeCheck())
+				{
+					Undo.RecordObject(bh, "Change DragPoint Position to " + dpos.ToString());
+					dpos = wlMat.MultiplyPoint(dpos);
+					dpos -= offset;
+					dpos.z -= dpoint.CalcHeight;
+					dpoint.Vertex = dpos.ToVertex3D();
+
+					if (editable != null)
+					{
+						editable.MeshDirty = true;

```

Comment:
Setting this flag should happen before the undo.recordobject. That's how I solved the mesh rebuilds for the other inspectors, basically the idea is the mesh is "dirty" in the undo stack, so as soon as the object is reverted that state it'll trigger a mesh rebuild.

Reply:
Was wondering that :)